### ISSUE TITLE
Modulepreload does not work on client navigation

### ISSUE THREAD

**Issue by celebro** [remix-run/remix#6657]
### What version of Remix are you using?

1.17.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Open https://remix.run/ and browser network devtools. Navigate to blog. (I noticed this on on my own application running remix 1.17.0, but remix.run website exhibits the same problem)

### Expected Behavior

All javascript modules / chunks should preload at start of navigation.

### Actual Behavior

Only main route javascript loads, additional chunks only start loading after main javascript completes, despite all assets being available in the manifest. Preloading only works on initial page load, but not on client navigation.

![Screenshot 2023-06-21 at 09 56 10](https://github.com/remix-run/remix/assets/860165/fb4b672e-275b-4090-8528-b601906ef5b4)

From quick search it looks like the change was introduced here: https://github.com/remix-run/remix/commit/f717d02e#diff-594d4b6ed5bd4abff7e9319566b5505b6a85753e665ff4d2113e888384a98377R1090 ?

**Comment by celebro** [comment]
Yeah, you're right, I checked out older version and it never worked, I guess I just assumed based on initial page load it should. Should I open a feature request instead?

**Comment by brophdawg11** [comment]
Yeah I can just convert this issue to a Proposal Discussion
